flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarDeserializationSchemaTest.java

```diff
@@ -108,6 +122,243 @@ void createFromFlinkTypeInformation() throws Exception {
         assertEquals(collector.result, "test-content");
     }
 
+    @Test
+    void primitiveStringPulsarSchema() {
+        final String topicName =
+                "primitiveString-" + ThreadLocalRandom.current().nextLong(0, Long.MAX_VALUE);
+        operator().createTopic(topicName, 1);
+        String expectedMessage = randomAlphabetic(10);
+        operator()
+                .sendMessage(
+                        TopicNameUtils.topicNameWithPartition(topicName, 0),
+                        Schema.STRING,
+                        expectedMessage);
+        PulsarSource<String> source = createSource(topicName, pulsarSchema(Schema.STRING));
+        assertThatCode(() -> runPipeline(source, expectedMessage)).doesNotThrowAnyException();
```

Comment:
Here I thought no exception could prove the deserialization is ok, do we need to verify that the records are intact as well ?

Reply:
already did lol